chains/evm/contracts/Proxy.sol

```diff
@@ -3,29 +3,29 @@ pragma solidity ^0.8.24;
 
 import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
 
-/// @notice VerifierProxy enables upgrades to Cross-Chain Verifiers (CCVs) without breaking existing references in token pools, receivers, and apps.
+/// @notice Proxy enables upgrades to Cross-Chain Verifiers (CCVs) and Executors without breaking existing references in token pools, receivers, and apps.
 /// The address of this contract will be referenced in the following places:
-///   - Senders of messages will specify required and optional CCVs as part of ccipSend extraArgs.
+///   - Senders of messages can specify required and optional CCVs + a desired executor as part of ccipSend extraArgs.
 ///   - Token pools will specify required CCVs on both source and destination.
 ///   - Receiver contracts will specify required and optional CCVs on destination.
-///   - OnRamp will specify default and mandated CCVs for each destination.
+///   - OnRamp will specify default and mandated CCVs + a default executor for each destination.
 ///   - OffRamp will specify default and mandated CCVs for each source.
-/// Each of these references should be to a VerifierProxy contract, not a Verifier contract directly.
-/// @dev On source, the OnRamp will forward requests (i.e. getFee, forwardToVerifier) through this contract to the required Verifier.
-/// The same applies on destination. The OffRamp will forward requests (i.e. verifyMessage) through this contract to the required Verifier.
-/// To support this proxy, all future Verifier interfaces must have originalCaller defined as the first arg to each method.
-contract VerifierProxy is Ownable2StepMsgSender {
+/// Each of these references should be to a Proxy contract, not a CCV / Executor directly.
+/// @dev On source, the OnRamp will forward requests (i.e. getFee, forwardToVerifier) through this contract to the required CCV / Executor.
+/// The same applies on destination. The OffRamp will forward requests (i.e. verifyMessage) through this contract to the required CCV.
+/// To support this proxy, all relevant interfaces must define originalCaller as the first arg to each method.
+contract Proxy is Ownable2StepMsgSender {
```

Comment:
I think its best if the name references the fact it overrides the first param with sender. Something like ProxyWithSender